controllers/cinder_controller.go

```diff
@@ -845,6 +845,24 @@ func (r *CinderReconciler) transportURLCreateOrUpdate(
 }
 
 func (r *CinderReconciler) apiDeploymentCreateOrUpdate(ctx context.Context, instance *cinderv1beta1.Cinder) (*cinderv1beta1.CinderAPI, controllerutil.OperationResult, error) {
+
+	apiTemplate := instance.Spec.CinderAPI
+	cinderAPISpec := cinderv1beta1.CinderAPISpec{
+		ContainerImage:         apiTemplate.ContainerImage,
+		Replicas:               apiTemplate.Replicas,
+		Debug:                  apiTemplate.Debug,
+		CustomServiceConfig:    apiTemplate.CustomServiceConfig,
```

Comment:
I think you missed the CustomServiceConfigSecrets field (here, and in the backup, scheduler and volume functions as well).

Reply:
Good catch and sorry, I missed that! Adding the missing `CustomServiceConfigSecrets` in the next PS!

Reply:
Done, feel free to resolve the conversation if this part it's ok.